tests/reftests/clean.test

```diff
@@ -109,3 +110,175 @@ Remaining files:
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[default] synchronised from file://${BASEDIR}/REPO
+Now run 'opam upgrade' to apply any package updates.
+### opam install waste
+The following actions will be performed:
+=== install 1 package
+  - install waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved waste.1  (file://${BASEDIR}/waste.tgz)
+-> installed waste.1
+Done.
+### ls OPAM/clean/.opam-switch/sources/
+waste.1
+### opam remove waste
+The following actions will be performed:
+=== remove 1 package
+  - remove waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   waste.1
+Done.
+### ls OPAM/clean/.opam-switch/sources
+### opam install waste
+The following actions will be performed:
+=== install 1 package
+  - install waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved waste.1  (cached)
+-> installed waste.1
+Done.
+### opam reinstall waste
+The following actions will be performed:
+=== recompile 1 package
+  - recompile waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved waste.1  (cached)
+-> removed   waste.1
+-> installed waste.1
+Done.
+### ls OPAM/clean/.opam-switch/sources
+### opam remove waste
+The following actions will be performed:
+=== remove 1 package
+  - remove waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   waste.1
+Done.
+### opam install ./waste
+waste is now pinned to file://${BASEDIR}/waste (version 1)
+The following actions will be performed:
+=== install 1 package
+  - install waste 1 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved waste.1  (file://${BASEDIR}/waste)
+-> installed waste.1
+Done.
+### ls OPAM/clean/.opam-switch/sources
+waste
+### opam unpin waste
+Ok, waste is no longer pinned to file://${BASEDIR}/waste (version 1)
+The following actions will be performed:
+=== recompile 1 package
+  - recompile waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved waste.1  (cached)
+-> removed   waste.1
+-> installed waste.1
+Done.
+### opam remove waste
+The following actions will be performed:
+=== remove 1 package
+  - remove waste 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   waste.1
+Done.
+### opam pin waste --dev
+[waste.1] synchronised (file://${BASEDIR}/waste)
+waste is now pinned to file://${BASEDIR}/waste (version 1)
+
+The following actions will be performed:
+=== install 1 package
+  - install waste 1 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved waste.1  (file://${BASEDIR}/waste)
+-> installed waste.1
+Done.
+### ls OPAM/clean/.opam-switch/sources
+waste
+### opam unpin waste
```

Comment:
It feels like there should be more `ls` checks after these tests, confirming that exactly when the `waste` directory should still be there?